sklearn/neural_network/tests/test_elm.py

```diff
@@ -0,0 +1,393 @@
+
+    (n_samples, n_classes) = y.shape[0], np.unique(y).size
+
+    assert_equal(y_proba.shape, (n_samples, n_classes))
+
+
+def test_recursive_and_standard():
+    """Test that recursive lsqr return the same result as standard lsqr."""
+    batch_size = 50
+    for dataset, class_weight in product(classification_datasets.values(),
+                                         [None, 'auto']):
+        X, y = dataset
+        elm_standard = ELMClassifier(class_weight=class_weight,
+                                     random_state=random_state)
+        elm_recursive = ELMClassifier(class_weight=class_weight,
+                                      random_state=random_state,
+                                      batch_size=batch_size)
+        elm_standard.fit(X, y)
+        elm_recursive.fit(X, y)
+
+        pred1 = elm_standard.predict(X)
+        pred2 = elm_recursive.predict(X)
+
+        assert_array_equal(pred1, pred2)
+        assert_greater(elm_standard.score(X, y), 0.95)
+
+
+def test_sample_weight_elm():
+    """Smoke test - AdaBoostClassifier should work with ELMClassifer."""
+    X = Xdigits_binary[:50]
+    y = ydigits_binary[:50]
+
+    elm = ELMClassifier(n_hidden=20)
+    clf = AdaBoostClassifier(n_estimators=3, base_estimator=elm)
+    clf.fit(X, y)
+    assert_greater(clf.score(X, y), 0.9)
+
+
+def test_sparse_matrices():
+    """Test that sparse and dense input matrices yield equal output."""
+    X = Xdigits_binary[:50]
+    y = ydigits_binary[:50]
+    X = csr_matrix(X)
+    n_hidden = 15
+    batch_size = 10
+
+    # Standard ELM
+    elm = ELMClassifier(random_state=1, n_hidden=n_hidden)
+    # Batch based
+    elm_batch_based = ELMClassifier(random_state=1, n_hidden=n_hidden,
+                                    batch_size=10)
+    # ELM for partial fitting
+    elm_parital = ELMClassifier(random_state=1, n_hidden=n_hidden)
+    # Train classifiers
+    elm.fit(X, y)
+    elm_batch_based.fit(X, y)
+
+    for batch_slice in gen_batches(X.shape[0], batch_size):
+        elm_parital.partial_fit(X[batch_slice], y[batch_slice])
+
+    # Get decision scores
+    y_pred = elm.decision_function(X)
+    y_pred_batch_based = elm_batch_based.decision_function(X)
+    y_pred_partial = elm_parital.decision_function(X)
+
+    # The prediction values should be the same
+    assert_almost_equal(y_pred, y_pred_batch_based)
+    assert_almost_equal(y_pred_batch_based, y_pred_partial)
+
+
+def test_verbose():
+    """Test whether verbose works as intended."""
+    X = Xboston
+    y = yboston
+
+    elm_fit = ELMRegressor(verbose=True)
+    elm_batch_fit = ELMRegressor(verbose=True, batch_size=50)
+    for elm in [elm_fit, elm_batch_fit]:
+        old_stdout = sys.stdout
+        sys.stdout = output = StringIO()
+
+        elm.fit(X, y)
+        sys.stdout = old_stdout
+
+        assert_not_equal(output.getvalue(), '')
+
+
+def test_warmstart():
+    """Tests that warm_start reuses past solution only if set to true."""
```

Comment:
I feel this test should also test that the hidden to output weights are kept / updated. They should give the same results as if presented the whole dataset, right?